nbterm/tests/test_cli.py

```diff
@@ -0,0 +1,22 @@
+from pathlib import Path
+
+import pytest
+
+from nbterm.nbterm import default_save_path
+
+
+@pytest.mark.parametrize(
+    "notebook_name, save_name",
+    [
+        ("foo", "foo_run"),
+        ("foo.ipynb", "foo_run.ipynb"),
+        ("foo_.ipynb", "foo__run.ipynb"),
+        ("foo_run.ipynb", "foo_run_run.ipynb"),
+    ],
+)
+def test_default_save_path(notebook_name: str, save_name: str, tmp_path: Path) -> None:
+    assert default_save_path(tmp_path / notebook_name) == tmp_path / save_name
+
+
+def test_default_save_path_with_no_notebook_is_in_cwd() -> None:
+    assert Path.cwd() in default_save_path(None).parents
```

Comment:
What is this testing?

Reply:
Hmm, I think this is the root of my confusion re: this and also your comment here https://github.com/davidbrochart/nbterm/pull/34/files#r631319319

The way I was thinking about this is that `nb_path` should only be non-`None` if the notebook is being read from a file on disk. Otherwise, it should be `None`. I think in the current version, `nb_path` is serving two roles: sometimes it's a directory path, in which case it's used as the kernel working directory, and sometimes it's the path to an actual notebook file.

Perhaps `kernel_cwd` should be an argument to the `Notebook` constructor, to decouple these two usages? Then the `notebook_path` CLI arg can be processed outside of `Notebook.__init__`.

I guess I'm just not sure what `Notebook.nb_path` should be for various ways of invoking the CLI... the cases I can think of are (with my proposed set of values):
- no path given (`nb_path = None`, `kernel_cwd = Path.cwd()`, `save_path = default_save_path(None)`)
- a directory that already exists (`nb_path = None`, `kernel_cwd = notebook_path`, `save_path = default_save_path(notebook_path)`)
- an existing notebook (`nb_path = notebook_path`, `kernel_cwd = notebook_path.parent`, `save_path = default_save_path(notebook_path.parent)`)
- a notebook that doesn't exist yet, i.e., a path in a directory that does exist but the path itself does not (`nb_path = None`, `kernel_cwd = notebook_path.parent`, `save_path = default_save_path(notebook_path.parent)`)

And of course `save_path` can be overwritten by the CLI option.

Does that sound right?